### PR TITLE
ipq40xx: add e2600ac c1/c2 to DSA

### PR DESCRIPTION
**Pull request by zhangsd20**
1. Convert E2600ac c1/c2 to DSA and enable it.
2. Modify the kernel size(E2600AC c1) to fit the new kernel-5

Signed-off-by: 张鹏 <sd20@qxwlan.com>

### CODE REVIEW

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -90,6 +90,12 @@ ipq40xx_setup_interfaces()
 	plasmacloud,pa2200)
 		ucidef_set_interfaces_lan_wan "ethernet1" "ethernet2"
 		;;
+	qxwlan,e2600ac-c1)
+		ucidef_set_interfaces_lan_wan "lan1" "eth1"
+		;;
+	qxwlan,e2600ac-c2)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" "eth1"
```
Comment (Ansuel) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network:97]: eth1? is this correct?

**Comment by Ansuel** [comment]
Please add a commit description of what you are doing.

Also please split this in 3 commits, one for enlarging the kernel size and 2for enabling the device and the dsa configuration for each one...

**Comment by Ansuel** [comment]
(you can have multiple commit in one pr)

**Comment by zhangsd20** [comment]
Ok,thanks.


Sorry I'm trying how to do it. I haven't done anything like this before











在2023年02月22 09时20分，"Christian ***@***.***>写道：


(you can have multiple commit in one pr)

**Comment by Ansuel** [comment]
@zhangsd20 Hi, why you are closing this? It would be good to continue things here.

Anyway commit should be imperative.
I would ask you to change the first commit to

`ipq40xx: enlarge Qxwlan E2600AC c1 kernel size to 5120k`

and as commit description

```
Kernel 5.15 needs more space as the kernel is now bigger than the current limit.
Enlarge kernel size from 4096k to 5120k.
```

With this change this is ready and i will merge it.

**Comment by zhangsd20** [comment]
I'm sorry. Since I saw that the author's name did not match Signed-off-by,

I wanted to fix this issue. But when operating locally, the wrong operation was carried out,

resulting in some problems. I can only recreate a PR











在2023年02月22 21时57分，"Christian ***@***.***>写道：


@zhangsd20 Hi, why you are closing this? It would be good to continue things here.
Anyway commit should be imperative.
I would ask you to change the first commit to
ipq40xx: enlarge Qxwlan E2600AC c1 kernel size to 5120k
and as commit description
Kernel 5.15 needs more space as the kernel is now bigger than the current limit.
Enlarge kernel size from 4096k to 5120k.With this change this is ready and i will merge it.

**Comment by zhangsd20** [comment]
If this PR can be merged, then I will remove the new PR.

Thank you very much!

**Comment by zhangsd20** [comment]
If this PR can be merged, then I will remove the new PR.

Thank you very much!

**Comment by robimarko** [comment]
@Ansuel I posted comments in the new PR, this is just confusing.

So, to make the same questions available here:
1. What is the bootloader limit for kernel size, or is there even one?
Cause, if there is none there is no point in defining the kernel limit, and if there is one just set it now and not update every time the building breaks.

2. eth1 as the name makes no sense, is it printed on the device or?
If not, then make it `wan` or `lan2`

**Comment by robimarko** [comment]
You dont need the commit enlarging kernel size at all.

Also, the name and description for the `KERNEL_SIZE` removal are not really great.
It should probably be more like this:
```
ipq40xx: e2600ac-c1: remove KERNEL_SIZE

Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.
```

**Comment by zhangsd20** [comment]
I've tested it, it is fine.
I added a new commit as you suggested











在2023年02月23 19时55分，"Robert ***@***.***>写道：


You dont need the commit enlarging kernel size at all.
Also, the name and description for the KERNEL_SIZE removal are not really great.
It should probably be more like this:
ipq40xx: e2600ac-c1: remove KERNEL_SIZE

Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.—

**Pull request closed without merging**